plugin/src/main/java/com/denizenscript/denizen/objects/LocationTag.java

```diff
@@ -4265,6 +4267,97 @@ else if (material.hasModernData() && material.getModernData() instanceof org.buk
+            // Alternatively, you can specify a custom structure from a datapack, plugin, etc. as a namespaced key.
+            // See also <@link tag LocationTag.find_structure_type> to find structures by type,
+            // and <@link tag server.structures> for all structures currently available on the server.
+            // -->
+            tagProcessor.registerTag(LocationTag.class, MapTag.class, "find_structure", (attribute, object, input) -> {
+                ElementTag structureName = input.getRequiredObjectAs("structure", ElementTag.class, attribute);
+                ElementTag radius = input.getRequiredObjectAs("radius", ElementTag.class, attribute);
+                if (structureName == null || radius == null) {
+                    return null;
+                }
+                org.bukkit.generator.structure.Structure structure = Registry.STRUCTURE.get(Utilities.parseNamespacedKey(structureName.asString()));
+                if (structure == null) {
+                    attribute.echoError("Invalid structure specified: " + structureName + '.');
+                    return null;
+                }
+                if (!radius.isInt()) {
+                    attribute.echoError("Invalid radius '" + radius + " specified': must be a number.");
+                    return null;
+                }
+                ElementTag unexplored = input.getElement("unexplored", "false");
+                if (!unexplored.isBoolean()) {
+                    attribute.echoError("Invalid 'unexplored' value '" + unexplored + "' specified: must be a boolean.");
+                    return null;
+                }
+                StructureSearchResult searchResult = object.getWorld().locateNearestStructure(object, structure, radius.asInt(), unexplored.asBoolean());
+                return searchResult != null ? new LocationTag(searchResult.getLocation()) : null;
+            });
+
+            // <--[tag]
+            // @attribute <LocationTag.find_structure_type[type=<type>;radius=<#>(;unexplored=<true/{false}>)]>
+            // @returns MapTag
+            // @warning See <@link language Structure lookups> for potential issues/edge cases in structure lookups.
+            // @group finding
+            // @description
+            // Finds the closet structure of the given structure type within the specified chunk radius, optionally only searching for unexplored ones.
+            // See <@link url https://hub.spigotmc.org/javadocs/spigot/org/bukkit/generator/structure/StructureType.html> for all available structure types.
+            // The returned map, if any, has 2 keys:
+            // - "location", <@link ObjectType LocationTag> of the structure found.
+            // - "structure", <@link ObjectType ElementTag> of the type of structure found.
+            // See also <@link tag LocationTag.find_structure> to find specific structures instead of looking them up by type.
+            // -->
```

Comment:
These tags need `@example`s with real structure/structure type names, and to generally be more clear at-a-glance in the meta what the difference between a 'structure' and 'structure type' is.

... Also why is there a 'structures' list tag but no structure_type list tag? It seems like the old 'structure_type' is the new 'structure' but there's also a new 'structure_type' that is a distinct concept?